src/PostgREST/App.hs

```diff
@@ -147,7 +147,7 @@ app dbstructure conf reqBody req =
       -- select * from public.proc(a := "foo"::undefined) where whereT limit limitT
 
     ([], _) -> do
-      body <- encode <$> tables (cs schema)
+      body <- encode <$> accessibleTables (filter ((== cs schema) . tableSchema) allTabs)
```

Comment:
`dbTables` is initialized with all tables in the database except the ones in the `pg_catalog` and `information_schema` schemas. This is how I merged #342, we prefetch all the tables and then filter every request which ones the user actually has access to (vs selecting only the ones the user has access to from the outset).